async-stream-impl/src/lib.rs

```diff
@@ -160,7 +160,7 @@ impl VisitMut for Scrub<'_> {
                 } = expr;
 
                 let attr = attrs.pop().unwrap();
-                if let Err(e) = syn::parse2::<syn::parse::Nothing>(attr.tokens) {
+                if let Err(e) = attr.parse_args::<syn::parse::Nothing>() {
```

Comment:
Btw, this attribute is intended for internal use, so this case may not need to be checked.

(In the first version of PR that implemented the `for await` syntax, [the attribute was a public API](https://github.com/tokio-rs/async-stream/pull/2/commits/c1d4aefb3c75b1df610348e110d181e4f22a6e07#diff-8cad82973f9069baf2dd85a8e2fc20641586eafd2a8cd1174ea08603386eebec), so this check is a remnant of that.)

Reply:
Ah I see! I've removed the check in 07747d833b726